wtform_fields.py

```diff
@@ -0,0 +1,24 @@
+from flask_wtf import FlaskForm
+from wtforms import StringField, PasswordField, SubmitField
+from wtforms.validators import InputRequired, Length, EqualTo, ValidationError
+from models import User
+
+class RegistrationForm(FlaskForm):
+	""" Registration form """
+
+	username = StringField('username_label', 
+		validators=[InputRequired(message="Username required"), 
+		Length(min=4, max=25, message="Username must be between 4 and 25 charachters")])
+	password = PasswordField('password_label', 
+		validators=[InputRequired(message="Password required"), 
+		Length(min=4, max=25, message="Password must be between 4 and 25 charachters")])
+	confirm_pswd = PasswordField('confirm_pswd_label', 
+		validators=[InputRequired(message="Password confirmation required"), 
+		EqualTo('password', message="Passwords must match")])
+	submit_button = SubmitField('Create')
+
+	# Custom validator for the username
+	def validate_username(self, username):
+		user_object = User.query.filter_by(username=username.data).first();
+		if user_object:
+			raise ValidationError("Username already exists. Please choose another username")
```

Comment:
This is a security issue. Never return messages as ""Username already exists" or "Wrong password!". This gives the end user ability to hack your software. If this user is already existing then we can confirm that  certain username exists.  for example "shristov". Then  we could try username shristov with brute force attack or even better with phishing. That's why it is returned always "Wrong username or password!" and nothing more or less. Look at google or other big platforms

Reply:
This is only the registration page so your password really can`t be incorrect. All big platforms at the stage of registration are informing you that the username is already taken.
For example (gmail.com):

![taken-username](https://user-images.githubusercontent.com/25925936/76296677-34a92f80-62bf-11ea-9b29-eaee71e8cf0e.png)

Reply:
This is only the registration page so your password really can`t be incorrect. All big platforms at the stage of registration are informing you that the username is already taken.
For example (gmail.com):

![already-taken](https://user-images.githubusercontent.com/25925936/76297252-3c1d0880-62c0-11ea-91d3-ea58133fab2d.png)